gtests/net/packetdrill/run_packet.c

```diff
@@ -371,6 +371,31 @@ static struct socket *find_connect_for_live_packet(
 	return socket;
 }
 
+static struct socket *find_mp_join_for_live_packet(
+	struct state *state, struct packet *packet,
+	enum direction_t *direction)
+{
+	struct tuple tuple;
+	struct socket *socket = state->socket_under_test;	/* shortcut */
+	if (!socket)
+		return NULL;
+
+	*direction = DIRECTION_INVALID;
+	get_packet_tuple(packet, &tuple);
```

Comment:
small detail: even if your modification makes more sense, may you reorder what is above to imitate what is done in find_connect_for_live_packet()?


	struct tuple tuple;
	get_packet_tuple(packet, &tuple);

	*direction = DIRECTION_INVALID;
	struct socket *socket = state->socket_under_test;	/* shortcut */
	if (!socket)
		return NULL;

Just in case we need to adapt MPTCP code after a modification from upstream: easier if the code looks the same to easily spot the differences :)

Reply:
ah, right. Indeed, the bug was that MP_JOIN syn packets triggered by "subflow" endpoints weren't processed at all, because they had DIRECTION_INVALID. state->socket_under_test can be NULL on (rare, pathologic) cases but it's better to fix this. Thanks for noticing

Reply:
while at it, do you mind keeping the exact same order as in `find_connect_for_live_packet()` to be closer to the upstream code to ease the adaptation of the MPTCP code if the upstream one change? :)

So just move `get_packet_tuple()` and `*direction` a couple of lines before? :)

But a detail, at least here we avoid an issue ;-)